src/app-admin.js

```diff
@@ -0,0 +1,491 @@
+  if (!user) return;
+
+  const newUser = req.body;
+  const { id: userIdInDb } = getUserByUsername(newUser.userName);
+
+  if (!validateUserInput(res, newUser)) return;
+  if (userIdInDb) {
+    sendErrorResponse(
+      res,
+      400,
+      'user-already-exists',
+      `User ${newUser.userName} already exists`,
+    );
+    return;
+  }
+
+  const userId = uuid.v4();
+  let displayName = newUser.displayName || null;
+  let enabled = newUser.enabled ? 1 : 0;
+
+  getAccountDb().mutate(
+    'INSERT INTO users (id, user_name, display_name, enabled, master) VALUES (?, ?, ?, ?, 0)',
+    [userId, newUser.userName, displayName, enabled],
+  );
+
+  getAccountDb().mutate(
+    'INSERT INTO user_roles (user_id, role_id) VALUES (?, ?)',
+    [userId, newUser.role],
+  );
+
+  res.status(200).send({ status: 'ok', data: { id: userId } });
+});
+
+app.patch('/users', async (req, res) => {
+  const user = await getAdminSessionFromRequest(req, res);
+  if (!user) return;
+
+  const userToUpdate = req.body;
+  const { id: userIdInDb } =
+    getAccountDb().first('SELECT id FROM users WHERE id = ?', [
+      userToUpdate.id,
+    ]) || {};
+
+  if (!validateUserInput(res, userToUpdate)) return;
+  if (!userIdInDb) {
+    sendErrorResponse(
+      res,
+      400,
+      'cannot-find-user-to-update',
+      `Cannot find ${userToUpdate.userName} to update`,
+    );
+    return;
+  }
+
+  let displayName = userToUpdate.displayName || null;
+  let enabled = userToUpdate.enabled ? 1 : 0;
+
+  getAccountDb().mutate(
+    'UPDATE users SET user_name = ?, display_name = ?, enabled = ? WHERE id = ?',
+    [userToUpdate.userName, displayName, enabled, userIdInDb],
+  );
+
+  getAccountDb().mutate('UPDATE user_roles SET role_id = ? WHERE user_id = ?', [
+    userToUpdate.role,
+    userIdInDb,
+  ]);
+
+  res.status(200).send({ status: 'ok', data: { id: userIdInDb } });
+});
+
+app.post('/users/delete-all', async (req, res) => {
+  const user = await getAdminSessionFromRequest(req, res);
+  if (!user) return;
+
+  const ids = req.body.ids;
+  let totalDeleted = 0;
+  ids.forEach((item) => {
+    const { id: masterId } =
+      getAccountDb().first('SELECT id FROM users WHERE master = 1') || {};
+
+    if (item === masterId) return;
+
+    getAccountDb().mutate('DELETE FROM user_roles WHERE user_id = ?', [item]);
+    getAccountDb().mutate('DELETE FROM user_access WHERE user_id = ?', [item]);
+    getAccountDb().mutate('UPDATE files set owner = ? WHERE owner = ?', [
+      masterId,
+      item,
+    ]);
+    const usersDeleted = getAccountDb().mutate(
+      'DELETE FROM users WHERE id = ? and master = 0',
+      [item],
+    ).changes;
+    totalDeleted += usersDeleted;
+  });
+
+  if (ids.length === totalDeleted) {
+    res
+      .status(200)
+      .send({ status: 'ok', data: { someDeletionsFailed: false } });
+  } else {
+    sendErrorResponse(res, 400, 'not-all-deleted', '');
+  }
+});
+
+app.get('/access', (req, res) => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏(applicable to the other endpoints too)
```suggestion
app.get('/users/access', (req, res) => {
```